src/connect.ts

```
@@ -0,0 +1,83 @@
+	publicKey: string;
+	redirectURL: string;
+	services: Services;
+	verifiedPublicKey: boolean = false;
+
+	constructor(publicKey: string, redirectURL: string, services: Services) {
+		Connect.validateRedirectURL(redirectURL);
+		Connect.validateInputServices(services);
+
+		this.publicKey = publicKey;
+		this.redirectURL = redirectURL;
+		this.services = services;
+	}
+
+	async generateURL() {
+		await Connect.validatePublicKey(this.publicKey);
+		const services = JSON.stringify(this.services)
+		return `${APP_CLIP_BASE_URL}?services=${services}&redirectUrl=${this.redirectURL}&publicKey=${this.publicKey}`
+	}
+
+	async outputQRCode() {
+		await Connect.validatePublicKey(this.publicKey);
+		const services = JSON.stringify(this.services)
+		const url = `${APP_CLIP_BASE_URL}?services=${services}&redirectUrl=${this.redirectURL}&publicKey=${this.publicKey}`
+		const qrCode = new QRCodeStyling(qrCodeStyle(url));
+
+		return qrCode
+	}
+
+	// static async create(publicKey: string, redirectURL: string, services: Services): Promise<Connect> {
+    //     await Connect.validatePublicKey(publicKey);
+    //     Connect.validateRedirectURL(redirectURL);
+	// 	Connect.validateInputServices(services);
+    //     return new Connect(publicKey, redirectURL, services);
+    // }
+
+	private static async validatePublicKey(publicKey: string): Promise<void> {
+		const publicKeyExists = await verifyPublicKey(publicKey);
+		if (!publicKeyExists) {
+			throw new Error('Public key does not exist');
+		}
+	}
+
+	private static validateInputServices(input: Services): void {
+		const validKeys = Object.values(SupportedServices);
+		type stringArray = string[];
+		let unsupportedServices:stringArray = []
+		for (const key in input) {
+			if (!validKeys.includes(key as SupportedServices)) {
+				unsupportedServices = [...unsupportedServices, key]
+			}
+		}
+
+		if (unsupportedServices.length > 0) {
+			throw new Error(`These services ${unsupportedServices.join(' ')} are unsupported`)
+		}
+	}
```

There should be a check to make sure the user doesn't pass only one service as an optional service i.e `{netflix: false}` should throw an error